### PR TITLE
Feature request: privacy statement and imprint

### LINKED ISSUE #297: Feature request: privacy statement and imprint
When using the admin frontend for more than a few admins (e.g. lecturers that upload videos directly from the admin ui), there has to be a way to supply a privacy statement and an imprint. Both should off course be configurable and being linked in the UI´s footer.

Backend issue: https://github.com/opencast/opencast/issues/5614

### PR DESCRIPTION
**Pull request by schuettloeffel-elsa**
closes #297

- Added Imprint and Privacy links to footer
- Implemented fetching of imprint and privacy html files
- Translated both terms to all languages
- Added test files
- also see backend PR: https://github.com/opencast/opencast/pull/5674

image

### CODE REVIEW

File: app/src/components/About.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			<Header />
+			<section className="action-nav-bar">
+				<MainNav isOpen={displayNavigation} toggleMenu={toggleNavigation} />
+				<nav>
+					<Link to="/about/imprint" className={cn({ active: location.pathname === "/about/imprint" })} onClick={() => { }}>{t("IMPRINT")}</Link>
+					<Link to="/about/privacy" className={cn({ active: location.pathname === "/about/privacy" })} onClick={() => { }}>{t("PRIVACY")}</Link>
+				</nav>
+			</section>
+			<div className="about">
+				<div dangerouslySetInnerHTML={{ __html: aboutContent }} ></div>
```
Comment (Arnei) [inline app/src/components/About.tsx:54]: In the case that there is not conent, the user is met with an empty gray block. Having some form of error message ("The imprint is unfortunately unavailable at this time") would be good.

File: app/src/styles/components/_components-config.scss [hunk 2]
```diff
@@ -40,6 +40,8 @@
 @import "stats";
 @import "popover";
 @import "collapsible-box";
+@import "collapsible-box";
```
Comment (Arnei) [inline app/src/styles/components/_components-config.scss:43]: Importing this twice is probably not intended?

**Review by Arnei — COMMENTED** [review]
Played around with this a bit and what can I say, it works :+1: I'll let others comment on whether the design of the pages is in line with the admin ui design docs, and instead raise some discussions points about technical stuff below.

I would prefer if we could avoid using user-generated HTML and JSON or some other form of string map instead. This would allow us more control in the frontend to ensure everything looks well, users don't have to fiddle around with HTML, and we could get some stuff like "Tel." or "Fax" translated in the frontend so users don't have to.
But I am also not at all knowledgable about the requirements for imprints and privacy statements, so if you say they absolutely require this degree of flexiblity I'm fine with it.

**Comment by schuettloeffel-elsa** [comment]
> I would prefer if we could avoid using user-generated HTML and JSON or some other form of string map instead. This would allow us more control in the frontend to ensure everything looks well, users don't have to fiddle around with HTML, and we could get some stuff like "Tel." or "Fax" translated in the frontend so users don't have to. But I am also not at all knowledgable about the requirements for imprints and privacy statements, so if you say they absolutely require this degree of flexiblity I'm fine with it.

In my experience at least the privacy statements need that degree of flexibility since they are highliy unique per organization and would be almost impossible to be generalized into a template of some kind (e.g. some organizations use kind of tables in their privacy statements to visualize data access).

File: app/src/components/About.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (schuettloeffel-elsa) [reply app/src/components/About.tsx:54]: Hm.. if we use plain html in the config files, I doubt there is an alternative to dangerouslySetInnerHTML(). But sanitizing it is still a good idea, gonna implement it.

File: app/src/styles/components/_components-config.scss (hunk 2 above)
Comment (schuettloeffel-elsa) [reply app/src/styles/components/_components-config.scss:43]: Thanks, will remove it.

**Comment by Arnei** [comment]
> > I would prefer if we could avoid using user-generated HTML and JSON or some other form of string map instead. This would allow us more control in the frontend to ensure everything looks well, users don't have to fiddle around with HTML, and we could get some stuff like "Tel." or "Fax" translated in the frontend so users don't have to. But I am also not at all knowledgable about the requirements for imprints and privacy statements, so if you say they absolutely require this degree of flexiblity I'm fine with it.
> 
> In my experience at least the privacy statements need that degree of flexibility since they are highliy unique per organization and would be almost impossible to be generalized into a template of some kind (e.g. some organizations use kind of tables in their privacy statements to visualize data access).

Unfortunate, but oh well, it's just like that sometimes :D. In that case I'm fine with approving this once you get around to the other comments.

File: app/src/components/Footer.tsx [hunk 3]
```diff
@@ -19,15 +22,20 @@ const Footer: React.FC = () => {
...
+					{/* Only render if a version is set */}
+					{!!user.ocVersion && (
+						<li>
+							Opencast {user.ocVersion.version}
+							{hasAccess("ROLE_ADMIN", user) && (
+								<span> - {user.ocVersion.buildNumber || "undefined"}</span>
+							)}
+						</li>
+					)}
+					<li><Link to="/about/imprint">{t("IMPRINT")}</Link></li>
```
Comment (gregorydlogan) [inline app/src/components/Footer.tsx:35]: I'm wondering if we should hide these by default, or at least provide a way to turn them off?  I'm sure the majority of institutions will (eventually) get to it, but for a lot of early adopters I can see not wanting boilerplate text linked from here.

**Comment by schuettloeffel-elsa** [comment]
Sorry for the delay, had lots of other stuff to do, I addressed all your concerns-

@Arnei: can you have a look now? 
@gregorydlogan: I implemented a switch to turn on/off both Footer Elements, you can configure it in `org.opencastproject.admin.display_about` in the me.json file.

**Review by ziegenberg — CHANGES_REQUESTED** [review]

File: app/src/i18n/org/opencastproject/adminui/languages/lang-de_DE.json [hunk 4]
```diff
@@ -2139,5 +2139,10 @@
 			"medium": "{{time, HH:mm:ss}}",
 			"full": "{{time, HH:mm [Uhr] z}}"
 		}
+	},
+	"ABOUT": {
+		"IMPRINT": "Impressum",
+		"PRIVACY": "Datenschutzerklärung",
+		"NOCONTENT": "Inhalt nicht verfügbar"
```
[the remark this answers is not in the document]
Comment (schuettloeffel-elsa) [reply app/src/i18n/org/opencastproject/adminui/languages/lang-de_DE.json:2146]: is this also true for new variables?

[the remark this answers is not in the document]
Comment (ziegenberg) [reply app/src/i18n/org/opencastproject/adminui/languages/lang-de_DE.json:2146]: Yes. Updates to language strings (inserts, updates, deletes) come via Crowdin.

[the remark this answers is not in the document]
Comment (schuettloeffel-elsa) [reply app/src/i18n/org/opencastproject/adminui/languages/lang-de_DE.json]: why didnt i knew that before translation to all these languages 😄

[the remark this answers is not in the document]
Comment (schuettloeffel-elsa) [reply app/src/i18n/org/opencastproject/adminui/languages/lang-de_DE.json]: removed all non en_us translations now

**Review by ziegenberg — APPROVED** [review]

File: app/src/components/About.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+					.then(response => {
+						setAboutContent(response.data);
+					})
+					.catch(error => {
+						console.error('Error while fetching data:', error);
+						setAboutContent(t("ABOUT.NOCONTENT").toString());
+					});
+			});
+
+	}, [location.pathname]); // Listen to changes in pathname
```
Comment (ziegenberg) [inline app/src/components/About.tsx:43]: `eslint` complains here:
```
React Hook useEffect has a missing dependency: 't'.
```
This warning is "fixed" this way in other places in the code the same way.

```suggestion
           // eslint-disable-next-line react-hooks/exhaustive-deps
	}, [location.pathname]); // Listen to changes in pathname
```

File: app/src/components/Footer.tsx [hunk 6]
```diff
@@ -5,12 +5,15 @@ import {
 } from "../selectors/userInfoSelectors";
 import { hasAccess } from "../utils/utils";
 import { useAppSelector } from "../store";
+import { Link, useNavigate } from "react-router-dom";
```
Comment (ziegenberg) [inline app/src/components/Footer.tsx:8]: `eslint` complains:
```
'useNavigate' is defined but never used
```
Can this go? Do we really need this?

```suggestion
import { Link } from "react-router-dom";
```

**Review by ziegenberg — CHANGES_REQUESTED** [review]

**Comment by ziegenberg** [comment]
I did a build run and found some warnings.

**Comment by schuettloeffel-elsa** [comment]
Thanks, commitet your suggestions

**Review by ziegenberg — APPROVED** [review]

**Pull request merged**